lib/factory_bot/definition_proxy.rb

```diff
@@ -170,5 +171,19 @@ def trait(name, &block)
     def initialize_with(&block)
       @definition.define_constructor(&block)
     end
+
+    private
+
+    def warn_static_attribute_deprecation(name, value)
+      attribute_caller = caller(2)
+
+      if attribute_caller[0].include?("method_missing")
+        attribute_caller = caller(3)
```

Comment:
We need to skip further up the callstack depending on whether the attribute was declared explicitly (`add_attribute(:x, "y")`) or implicitly (`x: "y"`)